src/app/dataset-view/dataset.component.ts

```diff
@@ -460,8 +457,8 @@ export class DatasetComponent
         const paramMap: ParamMap = this.activatedRoute.snapshot.paramMap;
         return {
             // Both parameters are mandatory in URL, router would not activate this component otherwise
-            accountName: paramMap.get(ProjectLinks.urlParamAccountName)!,
-            datasetName: paramMap.get(ProjectLinks.urlParamDatasetName)!,
+            accountName: paramMap.get(ProjectLinks.urlParamAccountName) || "",
```

Comment:
In this place we would like to throw a runtime error, if parameter is not present. If it is missing, it''s a guaranteed problem in our own logic, not because of user input ...

Since eslint really does not allow using non-null assertion operator !, let's define an utility helper function something like this:

```
function requireValue<T>(input: MaybeNull<T>){
  if (input === null) throw Error("value is required!");
  return input;
}
```

Reply:
Resolved